src/util/adb.js

```diff
@@ -213,6 +213,54 @@ export default class ADBUtils {
 
     return artifactsDir;
   }
+  fun(x: string) {
+    return x;
+  }
+  async checkArtifacts(deviceId: string): Promise<boolean> {
+    const {adbClient} = this;
+    await wrapADBCall(async () => {
+      await adbClient.readdir(deviceId, '/sdcard')
+        .then(function(dirs) {
+          // Synchronous, so we don't have to care about returning at the
+          // right time
+          dirs.forEach(function(dir) {
+            if (dir.isDir()) {
+              if (dir.name.startsWith('web-ext-artifacts-')) {
+                return false;
+              }
+            }
+          });
+        });
+    });
```

Comment:
When you `await` on a call to a function that return a promise you don't need to use a callback passed to `then` to get the resolved value, in other words instead of:

```
callToFunctionThatReturnAPromise().then(function (result) { ... });
```

**in an async function** you can just write it as:

```
const result = await callToFunctionThatReturnAPromise();
...
```

See https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await to learn more about how async/await allows you to rewrite Promise-based async code.

Reply:
Thanks for suggesting. Certainly wouldn't require await here.